htgr/assembly/tests

```diff
@@ -0,0 +1,51 @@
+[Tests]
+  [syntax_openmc]
+    type = RunApp
+    input = 'openmc.i'
+    check_input = true
+    executable_pattern = 'cardinal*'
+  []
+  [syntax_conduction]
+    type = RunApp
+    input = 'solid.i'
+    check_input = true
+    executable_pattern = 'cardinal*'
+  []
+  [syntax_thm]
+    type = RunApp
+    input = 'thm.i'
+    check_input = true
+    executable_pattern = 'cardinal*|thermal_hydraulics*'
+  []
+
+  [run_openmc]
+    type = RunApp
+    input = 'openmc.i'
+    cli_args = "MultiApps/active='' Transfers/active='' Executioner/num_steps=2"
+    executable_pattern = 'cardinal*'
+  []
+  [run_conduction]
+    type = RunApp
+    check_input = True
+    input = 'solid.i'
+    executable_pattern = 'cardinal*'
+  []
+  [run_thm]
+    type = RunApp
+    input = 'thm.i'
+    check_input = true
+    executable_pattern = 'cardinal*|thermal_hydraulics*'
+  []
+
+  # Check input generation
+  [run_python_generation]
+    type = RunCommand
+    command = 'mkdir test_dir; cp assembly.py test_dir/assembly.py; cd test_dir; python assembly.py'
+    required_python_packages = 'matplotlib uncertainties h5py'
```

Comment:
How does this test work? It needs the OpenMC Python API to work.

Reply:
I m going to debug it with the test suite, once the cardinal recipe is on for VTB.
It very likely wont run right now

Reply:
Would you please remind me the status of this PR? Now that we updated Cardinal, I think we should be able to use the same CIVET recipe for building Cardinal (we just won't be able to do the Python API from OpenMC, I think)